File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/dynamodb/AwsDynamoClient.java
package uk.nhs.nhsx.core.aws.dynamodb;

import com.amazonaws.services.dynamodbv2.document.DeleteItemOutcome;
import com.amazonaws.services.dynamodbv2.document.Item;

public interface AwsDynamoClient {
    void putItem (String tableName, Item item);
    Item getItem(String tableName, String hashKeyName, String hashKeyValue);
    DeleteItemOutcome deleteItem(String tableName, String hashKeyName, String hashKeyValue);
}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/core/aws/dynamodb/AwsDynamoClient.kt
package uk.nhs.nhsx.core.aws.dynamodb

import com.amazonaws.services.dynamodbv2.document.Item

interface AwsDynamoClient {
    fun getItem(
        tableName: String,
        hashKeyName: String,
        hashKeyValue: String
    ): Item?

    fun deleteItem(
        tableName: String,
        hashKeyName: String,
        hashKeyValue: String
    )
}
